homeassistant/components/unifiprotect/quality_scale.yaml

```diff
@@ -0,0 +1,164 @@
+  config-flow-test-coverage:
+    status: done
+    comment: test_config_flow.py provides comprehensive coverage of all config flow paths.
+  config-flow:
+    status: done
+    comment: Full config flow implementation in config_flow.py with VERSION = 2.
+  dependency-transparency:
+    status: done
+    comment: uiprotect library is open source on PyPI with tagged releases.
+  docs-actions:
+    status: done
+    comment: All 5 service actions documented at home-assistant.io/integrations/unifiprotect.
+  docs-high-level-description:
+    status: done
+    comment: Documentation describes UniFi Protect integration.
+  docs-installation-instructions:
+    status: done
+    comment: Prerequisites section with Local User setup, Camera Streams, API key creation.
+  docs-removal-instructions:
+    status: done
+    comment: Standard removal process applies, no special steps needed.
+  entity-event-setup:
+    status: done
+    comment: Uses async_on_remove pattern and entry.async_on_unload for cleanup.
+  entity-unique-id:
+    status: done
+    comment: All entities have unique_id based on device MAC and description key.
+  has-entity-name:
+    status: done
+    comment: BaseProtectEntity sets _attr_has_entity_name = True in entity.py.
+  runtime-data:
+    status: done
+    comment: Uses entry.runtime_data with typed UFPConfigEntry.
+  test-before-configure:
+    status: done
+    comment: Config flow validates connection via _async_get_nvr_data before entry creation.
+  test-before-setup:
+    status: done
+    comment: async_setup_entry calls protect.update() and raises ConfigEntryNotReady/AuthFailed.
+  unique-config-entry:
+    status: done
+    comment: Uses async_set_unique_id(nvr_data.mac) and _abort_if_unique_id_configured().
+
+  # Silver
+  action-exceptions:
+    status: done
+    comment: Services raise HomeAssistantError and ServiceValidationError appropriately.
+  config-entry-unloading:
+    status: done
+    comment: async_unload_entry implemented with platform unloading and data_service.async_stop().
+  docs-configuration-parameters:
+    status: done
+    comment: Options flow parameters documented (RTSP, realtime metrics, max media).
+  docs-installation-parameters:
+    status: done
+    comment: Hardware/software requirements and prerequisites fully documented.
+  entity-unavailable:
+    status: done
+    comment: WebSocket state handling marks entities unavailable on connection loss.
+  integration-owner:
+    status: done
+    comment: "@RaHehl listed as codeowner in manifest.json."
+  log-when-unavailable:
+    status: done
+    comment: _async_update_change logs connection loss and restoration in data.py.
+  parallel-updates:
+    status: done
+    comment: All platform files set PARALLEL_UPDATES = 0 for coordinator-based updates.
+  reauthentication-flow:
+    status: done
+    comment: async_step_reauth and async_step_reauth_confirm implemented in config_flow.py.
+  test-coverage:
```

Comment:
nit: `test_reboot_button`, `test_chime_button`, look similar and I think they can be parametrized